lib/cocoapods/command/repo/add.rb

```diff
@@ -83,8 +85,25 @@ def clone_repo
             Dir.chdir(config.repos_dir) do
               command = ['clone', @url]
               command << '--progress' if @progress
+              command << %w[--filter=blob:none --no-checkout --depth 1] if @sparse
```

Comment:
We should check with GitHub if adding `--depth 1` is going to be problematic here. (I expect it might be, given that they don't recommend that to folks here: https://github.blog/2020-01-17-bring-your-monorepo-down-to-size-with-sparse-checkout/)

Reply:
The `--depth 1` part cannot be removed, as cloning without it brings us back to the non-performant full clone scenario. 😭

The good news is that I think there's no need to worry about this. The cloning part doesn't even involve sparse checkouts at all, but separate git feature called [partial clones](https://about.gitlab.com/blog/2020/03/13/partial-clone-for-massive-repositories/). 

It basically allows you to perform a clone without bringing over all the objects. Additionally, it works great with shallow clones as it simply keeps the clone shallow and allows updating it forward without unshallowing.

That said, I'd love to run it by GH folks if you can point me to anyone, @endocrimes?